### PR TITLE
Fix UserInfo endpoint for deleted clients

### LINKED ISSUE #503: Crash at Introspection and UserInfo endpoints with token of deleted client
Use case:
1) Get access token for client
2) Delete client
3) Try to verify token (introspection endpoint) or get user info

### PR DESCRIPTION
**Pull request by tpazderka**
Close #503

- [ ] Any changes relevant to users are recorded in the `CHANGELOG.md`.
- [ ] The documentation has been updated, if necessary.
---

### CODE REVIEW

File: src/oic/oic/provider.py [hunk 1]
```diff
@@ -1202,7 +1202,9 @@ def _do_user_info(self, token, **kwargs):
         info = self.schema(**self._collect_user_info(session))
 
         # Should I return a JSON or a JWT ?
-        _cinfo = self.cdb[session["client_id"]]
+        _cinfo = self.cdb.get(session["client_id"])
+        if _cinfo is None:
+            return error_response('unauthorized_client', descr='Unknown client')
```
Comment (a-wakeel) [inline src/oic/oic/provider.py]: This could be handled inside the **except** block. Isn't?

Comment (a-wakeel) [inline src/oic/oic/provider.py]: This could be handled inside the **except** block. Isn't?

File: tests/test_oic_provider.py [hunk 2]
```diff
@@ -829,6 +829,32 @@ def test_userinfo_endpoint_authn(self):
         ident = OpenIDSchema().deserialize(resp.message, "json")
         assert _eq(ident.keys(), ['nickname', 'sub', 'name', 'email'])
 
+    def test_userinfo_endpoint_missing_client(self):
+        self.provider.cdb["unknownclient"] = {
+            "client_secret": "unknownclient",
+            "redirect_uris": [("http://localhost:8087/authz", None)],
+            "post_logout_redirect_uris": [("https://example.com/post_logout", None)],
+            "client_salt": "salted",
+            'response_types': ['code', 'token', 'code id_token', 'none', 'code token', 'id_token']
```
Comment (a-wakeel) [inline tests/test_oic_provider.py]: Is there any specific reason for using **single** quotes here, as **double** quotes are used in other key pairs?

Comment (a-wakeel) [reply tests/test_oic_provider.py]: Same applies for other usages

File: src/oic/oic/provider.py (hunk 1 above)
Comment (tpazderka) [reply src/oic/oic/provider.py]: Yes, but not all implementations of ClientDatabase raise `KeyError`, so I went with safer approach.

File: tests/test_oic_provider.py (hunk 2 above)
Comment (tpazderka) [reply tests/test_oic_provider.py]: No specific. We do have an inconsistent quoting style :(

I will match the quotes inside a test to match.

**Pull request merged**